iotlabsshcli/open_a8.py

```diff
@@ -59,143 +57,115 @@ def _nodes_grouped(nodes):
 _MAKE_EXECUTABLE_CMD = 'chmod +x {}'
 _RUN_SCRIPT_CMD = 'screen -S {screen} -dm bash -c \"{path}\"'
 _QUIT_SCRIPT_CMD = 'screen -X -S {screen} quit'
+_REMOTE_A8_DIR = 'A8/.iotlabsshcli'
 
 
 def flash_m3(config_ssh, nodes, firmware, verbose=False):
-    """Flash the firmware of M3 of open A8 nodes."""
-    # Configure ssh and remote firmware names.
+    """ Flash the firmware of M3 co-microcontroller """
+    failed_hosts = []
+    # configure ssh and remote firmware names.
     groups = _nodes_grouped(nodes)
     ssh = OpenA8Ssh(config_ssh, groups, verbose=verbose)
-    remote_fw = os.path.join('~/A8/.iotlabsshcli', os.path.basename(firmware))
-
-    # Create firmware destination directory
-    try:
-        ssh.run(_MKDIR_DST_CMD.format(os.path.dirname(remote_fw)),
```

Comment:
IIRC, the `A8/.iotlabsshcli` doesn't exist if the `iotlab-ssh` was never used by a user before. This is why you have to create in the first place.

Reply:
it's because we use the parallel-ssh native client now (instead of Paramiko)  and it creates the directory if it doesn't exist. So don't need to use run command